File: app/src/main/java/ai/rotor/rotorvehicle/ui/dash/Dash.java
package ai.rotor.rotorvehicle.ui.dash;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import ai.rotor.rotorvehicle.R;

public class Dash extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dash_activity);
        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.container, DashFragment.newInstance())
                    .commitNow();
        }
    }
}

File: app/src/main/java/ai/rotor/rotorvehicle/ui/dash/Dash.kt
package ai.rotor.rotorvehicle.ui.dash

import ai.rotor.rotorvehicle.R
import android.Manifest
import android.content.pm.PackageManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import androidx.lifecycle.LifecycleOwner

class Dash : AppCompatActivity(), LifecycleOwner {

    val REQUEST_CAMERA_CODE = 101

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.dash_activity)
        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                    .replace(R.id.container, DashFragment.newInstance())
                    .commitNow()
        }

        verifyCameraPermission()
    }

    private fun verifyCameraPermission() {
        if (ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.CAMERA), REQUEST_CAMERA_CODE)
        }
    }

}
